### PR TITLE
[Merged by Bors] - feat port Algebra.Field.Basic

### PR DESCRIPTION
**Pull request by riccardobrasca**
fc2ed6f838ce7c9b7c7171e58d78eaf7b438fb0e

### CODE REVIEW

File: Mathlib/Algebra/Field/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+theorem inv_sub_inv' {a b : K} (ha : a ≠ 0) (hb : b ≠ 0) : a⁻¹ - b⁻¹ = a⁻¹ * (b - a) * b⁻¹ := by
+  rw [mul_sub, sub_mul, mul_inv_cancel_right₀ hb, inv_mul_cancel ha, one_mul]
+
+theorem one_div_mul_sub_mul_one_div_eq_one_div_add_one_div (ha : a ≠ 0) (hb : b ≠ 0) :
+    1 / a * (b - a) * (1 / b) = 1 / a - 1 / b := by
+  rw [mul_sub_left_distrib (1 / a), one_div_mul_cancel ha, mul_sub_right_distrib, one_mul,
+    mul_assoc, mul_one_div_cancel hb, mul_one]
+
+-- see Note [lower instance priority]
+instance (priority := 100) DivisionRing.IsDomain : IsDomain K :=
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Field/Basic.lean:137]: Is the name here correct? Am I supposed to add `inst` or maybe to use the auto generated one?

Comment (semorrison) [inline Mathlib/Algebra/Field/Basic.lean:137]:
```suggestion
instance (priority := 100) DivisionRing.isDomain : IsDomain K :=
```

Comment (semorrison) [reply Mathlib/Algebra/Field/Basic.lean:137]: Similarly below.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**